addons/stock_account/models/account_invoice.py

```diff
@@ -0,0 +1,87 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+import logging
+
+from odoo import api, models
+
+_logger = logging.getLogger(__name__)
+
+
+class AccountInvoiceLine(models.Model):
+    _inherit = "account.invoice.line"
+
+    def _get_anglo_saxon_price_unit(self):
+        self.ensure_one()
+        return self.product_id.standard_price
+
+    def _get_price(self, inv, company_currency, i_line, price_unit):
+        if inv.currency_id.id != company_currency:
+            price = self.env['res.currency'].with_context({}, date=inv.date_invoice).compute(company_currency, inv.currency_id.id, price_unit * i_line.quantity)
+        else:
+            price = price_unit * i_line.quantity
+        return round(price, inv.currency_id.decimal_places)
+
+    def get_invoice_line_account(self, inv_type, product, fpos, company):
+        if company.anglo_saxon_accounting and inv_type in ('in_invoice', 'in_refund') and product and product.type == 'product':
+            accounts = product.product_tmpl_id.get_product_accounts(fiscal_pos=fpos)
+            return accounts['stock_input']
+        return super(AccountInvoiceLine, self).get_invoice_line_account(inv_type, product, fpos, company)
+
+
+class AccountInvoice(models.Model):
+    _inherit = "account.invoice"
+
+    @api.model
+    def invoice_line_move_line_get(self):
+        res = super(AccountInvoice,self).invoice_line_move_line_get()
+        if self.company_id.anglo_saxon_accounting:
+            if self.type in ('out_invoice','out_refund'):
```

Comment:
Use self.company_id.anglo_saxon_accounting and self.type in ('out_invoice','out_refund')